casper/src/main/scala/coop/rchain/casper/BlockMetadata.scala

```diff
@@ -1,15 +1,18 @@
 package coop.rchain.casper
 
 import com.google.protobuf.ByteString
-
 import coop.rchain.casper.Estimator.BlockHash
 import coop.rchain.casper.protocol.{BlockMessage, Justification}
+import coop.rchain.casper.util.ProtoUtil
 
 final case class BlockMetadata(
     blockHash: BlockHash,
     parents: List[BlockHash],
     sender: ByteString,
-    justifications: List[Justification]
+    justifications: List[Justification],
+    //todo think about whether this takes up too much space and if it needs to be kept elsewhere
```

Comment:
The calculation goes as follows

((((32 bits for validator + 64 bits for long) * N validators) / 8 bits per byte) / 1024*1024 bytes per MB) * B number of blocks

For N = 300 validators and B = 10k blocks, we have 34.33 MB. With a 10% overhead from the JVM, it is still in reason.

You can remove this TODO.

Reply:
Ok, thanks.